backends/advanced/src/advanced_omi_backend/controllers/system_controller.py

```
@@ -962,3 +1026,281 @@ async def validate_plugins_config_yaml(yaml_content: str) -> dict:
+        config: Configuration to test (same structure as update_plugin_config_structured)
+
+    Returns:
+        Test result with success status and details
+    """
+    try:
+        from advanced_omi_backend.services.plugin_service import discover_plugins, expand_env_vars
+
+        # Validate plugin exists
+        discovered_plugins = discover_plugins()
+        if plugin_id not in discovered_plugins:
+            raise ValueError(f"Plugin '{plugin_id}' not found")
```

Ruff (0.14.14)

Abstract `raise` to an inner function

(TRY301)